src/main/java/com/logsentinel/KeywordsExtractor.java

```diff
@@ -0,0 +1,8 @@
+package com.logsentinel;
+
+import java.util.List;
+
+public interface KeywordsExtractor {
```

Comment:
Given our other discussion, I think the interface could be called "EncryptingKeywordExtractor" and the implementation could be called "LuceneEncryptingKeywordExtractor"